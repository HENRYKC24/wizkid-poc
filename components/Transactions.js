import React from "react";
import {
  Image,
  StyleSheet,
  Text,
  View,
  ScrollView,
  TouchableWithoutFeedback,
  TouchableOpacity,
} from "react-native";
import transactions from "../utils/transactions";
import separateThousandsWithCommas from "../utils/separateThousandsWithCommas";

import group from "../assets/images/group.png";

const Transactions = () => {
  return (
    <ScrollView style={styles.container}>
      <View style={styles.transHeader}>
        <Text style={styles.tranText}>Transactions</Text>
        <TouchableOpacity>
          <Image style={styles.group} source={group} />
        </TouchableOpacity>
      </View>
      <ScrollView>
        {transactions.map((trans) => (
          <TouchableWithoutFeedback key={trans.id} style={styles.transBox}>
            <View style={styles.transactionItem}>
              <View style={styles.typeBox}>
                <Text style={styles.transType}>
                  {trans.type === "credit" ? "Received Money" : "Sent Money"}
                </Text>
                <Text style={styles.transWho}>{trans.who}</Text>
              </View>
              <View style={styles.transDetails}>
                <Text style={styles.transAmount}>
                  {trans.type === "credit" ? "+" : "-"}
                  {" $"}
                  {separateThousandsWithCommas(trans.amount.toFixed(2))}
                </Text>
                <Text style={styles.transDate}>{trans.date}</Text>
              </View>
            </View>
          </TouchableWithoutFeedback>
        ))}
      </ScrollView>
    </ScrollView>
  );
};

export default Transactions;

const styles = StyleSheet.create({
  container: {
    marginBottom: 250,
  },
  transBox: {
    // height: 200,
  },
  transHeader: {
    flexDirection: "row",
    justifyContent: "space-between",
    marginTop: 20,
    alignItems: "center",
  },
  tranText: {
    fontSize: 24,
    fontWeight: "400",
  },
  transactionItem: {
    flexDirection: "row",
    justifyContent: "space-between",
    marginTop: 20,
    alignItems: "center",
  },
  transType: {
    color: "#31456a",
    fontSize: 15,
    fontWeight: "400",
  },
  transWho: {
    color: "#7787A2",
    fontSize: 10,
    fontWeight: "400",
  },
  transAmount: {
    color: "#677895",
    fontSize: 16,
    fontWeight: "400",
  },
  transDate: {
    color: "#7787A2",
    fontSize: 10,
    fontWeight: "400",
  },
  transDetails: {
    display: "flex",
    alignItems: "flex-end",
  },
});
