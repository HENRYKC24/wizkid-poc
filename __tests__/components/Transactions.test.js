import React from "react";
import renderer from "react-test-renderer";

import Transactions from "../../components/Transactions";

describe("<Transactions />", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<Transactions />).toJSON();
    expect(tree.children.length).toBe(1);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Transactions />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
