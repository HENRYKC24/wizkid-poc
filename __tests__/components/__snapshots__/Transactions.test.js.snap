// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Transactions /> renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "marginBottom": 250,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "400",
          }
        }
      >
        Transactions
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
        />
      </View>
    </View>
    <RCTScrollView>
      <View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 20,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#31456a",
                  "fontSize": 15,
                  "fontWeight": "400",
                }
              }
            >
              Received Money
            </Text>
            <Text
              style={
                Object {
                  "color": "#7787A2",
                  "fontSize": 10,
                  "fontWeight": "400",
                }
              }
            >
              ridoyrock@gmail.com
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#677895",
                  "fontSize": 16,
                  "fontWeight": "400",
                }
              }
            >
              +
               $
              1,600.00
            </Text>
            <Text
              style={
                Object {
                  "color": "#7787A2",
                  "fontSize": 10,
                  "fontWeight": "400",
                }
              }
            >
              Sep 23, 2021
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 20,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#31456a",
                  "fontSize": 15,
                  "fontWeight": "400",
                }
              }
            >
              Sent Money
            </Text>
            <Text
              style={
                Object {
                  "color": "#7787A2",
                  "fontSize": 10,
                  "fontWeight": "400",
                }
              }
            >
              Airbnb Payment
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#677895",
                  "fontSize": 16,
                  "fontWeight": "400",
                }
              }
            >
              -
               $
              600.00
            </Text>
            <Text
              style={
                Object {
                  "color": "#7787A2",
                  "fontSize": 10,
                  "fontWeight": "400",
                }
              }
            >
              Sep 20, 2022
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 20,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#31456a",
                  "fontSize": 15,
                  "fontWeight": "400",
                }
              }
            >
              Received Money
            </Text>
            <Text
              style={
                Object {
                  "color": "#7787A2",
                  "fontSize": 10,
                  "fontWeight": "400",
                }
              }
            >
              Upwork Earning
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#677895",
                  "fontSize": 16,
                  "fontWeight": "400",
                }
              }
            >
              +
               $
              310.00
            </Text>
            <Text
              style={
                Object {
                  "color": "#7787A2",
                  "fontSize": 10,
                  "fontWeight": "400",
                }
              }
            >
              Sep 23, 2021
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 20,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#31456a",
                  "fontSize": 15,
                  "fontWeight": "400",
                }
              }
            >
              Received Money
            </Text>
            <Text
              style={
                Object {
                  "color": "#7787A2",
                  "fontSize": 10,
                  "fontWeight": "400",
                }
              }
            >
              ridoyrock@gmail.com
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#677895",
                  "fontSize": 16,
                  "fontWeight": "400",
                }
              }
            >
              +
               $
              1,600.00
            </Text>
            <Text
              style={
                Object {
                  "color": "#7787A2",
                  "fontSize": 10,
                  "fontWeight": "400",
                }
              }
            >
              Sep 23, 2021
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 20,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#31456a",
                  "fontSize": 15,
                  "fontWeight": "400",
                }
              }
            >
              Sent Money
            </Text>
            <Text
              style={
                Object {
                  "color": "#7787A2",
                  "fontSize": 10,
                  "fontWeight": "400",
                }
              }
            >
              Airbnb Payment
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#677895",
                  "fontSize": 16,
                  "fontWeight": "400",
                }
              }
            >
              -
               $
              600.00
            </Text>
            <Text
              style={
                Object {
                  "color": "#7787A2",
                  "fontSize": 10,
                  "fontWeight": "400",
                }
              }
            >
              Sep 20, 2022
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 20,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#31456a",
                  "fontSize": 15,
                  "fontWeight": "400",
                }
              }
            >
              Received Money
            </Text>
            <Text
              style={
                Object {
                  "color": "#7787A2",
                  "fontSize": 10,
                  "fontWeight": "400",
                }
              }
            >
              Upwork Earning
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#677895",
                  "fontSize": 16,
                  "fontWeight": "400",
                }
              }
            >
              +
               $
              310.00
            </Text>
            <Text
              style={
                Object {
                  "color": "#7787A2",
                  "fontSize": 10,
                  "fontWeight": "400",
                }
              }
            >
              Sep 23, 2021
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTScrollView>
`;
